sdk/src/binary.rs

```diff
@@ -196,6 +201,44 @@ impl VariableLengthBinary for Binary {
     }
 }
 
+impl Debug for Binary {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(f, "Binary(")?;
+        let mut iter = self.iter();
+        if let Some(x) = iter.next() {
+            write!(f, "{:?}", x)?;
+        }
+        for x in iter {
+            write!(f, ", {:?}", x)?;
+        }
+        write!(f, ")")?;
+        Ok(())
+    }
+}
+
+#[derive(Clone)]
+pub struct BinIter(Binary);
```

Comment:
Could you move this to under the IntoIterator impl? It might be helpful if the file is structured with these types in the same order as vec.rs, just for general readability.

Reply:
Done.